components/tikv_util/src/sys/thread.rs

```diff
@@ -264,10 +385,38 @@ mod tests {
     }
 
     #[test]
-    #[cfg(target_os = "linux")]
+    #[cfg(any(target_os = "linux", target_os = "macos"))]
     fn test_thread_ids() {
-        let ids: std::io::Result<Vec<_>> = thread_ids(process_id());
-        assert!(ids.is_ok());
-        assert!(!ids.unwrap().is_empty());
+        const THREAD_COUNT: usize = 10;
+        let (tx, rx) = crossbeam::channel::bounded(THREAD_COUNT);
+        let stop_threads = Arc::new(AtomicBool::new(false));
+
+        let threads = (0..THREAD_COUNT)
+            .map(|_| {
+                let tx = tx.clone();
+                let stop_threads = stop_threads.clone();
+                std::thread::spawn(move || {
+                    tx.send(thread_id()).unwrap();
+                    while !stop_threads.load(Ordering::Relaxed) {}
```

Comment:
It doesn't seem a good idea to let 10 threads watch a bool flag in a busy loop. I suggest adding use condvar or at least add some sleep.

Reply:
updated